velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -310,5 +309,65 @@ TEST_F(DateTimeFunctionsTest, dayOfMonth) {
   EXPECT_EQ(23, day(parseDate("2023-08-23")));
 }
 
+TEST_F(DateTimeFunctionsTest, dayOfWeekDate) {
+  const auto dayOfWeek = [&](std::optional<int32_t> date,
+                             const std::string& func) {
+    return evaluateOnce<int32_t, int32_t>(
+        fmt::format("{}(c0)", func), {date}, {DATE()});
+  };
+
+  for (const auto& func : {"dayofweek", "dow"}) {
+    EXPECT_EQ(std::nullopt, dayOfWeek(std::nullopt, func));
+    EXPECT_EQ(5, dayOfWeek(0, func));
+    EXPECT_EQ(4, dayOfWeek(-1, func));
+    EXPECT_EQ(7, dayOfWeek(-40, func));
```

Comment:
Can you add all the test here?https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala#L253